container/kubernetes/onboarding/deviceonboarding.py

```diff
@@ -0,0 +1,345 @@
+        # toolversion = valist[2]
+        deployurl = valist[3]
+        deploydir = valist[4]
+        deploygittag = valist[5]
+
+        # TestSensor doesn't have Machine Config so check and store
+        # relevant Machine Config data.
+        machineconfig_exist = True
+        if len(valist) > 6:
+            machineurl = valist[6]
+            machinedir = valist[7]
+            machinegittag = valist[8]
+        else:
+            machineconfig_exist = False
+
+        # Clear the every Device deployment data after parse and storage.
+        valist.clear()
+
+        # Preserve application dir for "update" and "delete" deployment
+        if os.path.exists(appname) and deploy_type == "create":
+            shutil.rmtree(appname)
+
+        # Local directory doesn't exists for "update" and "delete" deployment
+        # than it's same as "create" new deployment.
+        if not os.path.exists(appname):
+            print("Creating dir for application:", appname)
+            os.makedirs(appname)
+            git_clone_tempdir(deployurl, deploydir, deploygittag, appname, "d")
+            if machineconfig_exist:
+                git_clone_tempdir(machineurl, machinedir,
+                                  machinegittag, appname, "g")
+            # Deploying the Application pods
+            subprocess.call(['sh', 'kubernetes_deployment_creation.sh',
+                             nodename, appname, NAME_SPACE, deploy_type])
+            # Clean-up old local deployment directory.
+            if deploy_type == "delete":
+                shutil.rmtree(appname)
+        # Deployment action is "update" and local directory doesn't exists
+        # We need to run re-creation process.
+        elif deploy_type == "update":
+            # Function call for processing git repo and copyign relevant
+            # deployment-config *.yaml files.
+            git_clone_tempdir(deployurl, deploydir,
+                              deploygittag, appname, "d")
+            if machineconfig_exist:
+                # Function call for processing git repo and copyign relevant
+                # machine-config *.yaml files.
+                git_clone_tempdir(machineurl, machinedir,
+                                  machinegittag, appname, "g")
+            # Deploying the Application pods
+            subprocess.call(['sh', 'kubernetes_deployment_creation.sh',
+                             nodename, appname, NAME_SPACE, deploy_type])
+        # Deployment local directory exists than clean-up
+        # the deployment and local directory itself.
+        elif deploy_type == "delete":
+            print("Deleting Deployment:", appname)
+            deployname = appname + "/all.yaml"
+            subprocess.call(['kubectl', 'delete', '-f', deployname])
+            shutil.rmtree(appname)
+
+# pylint: disable=too-few-public-methods
+# Class will be expanded for adding more features.
+
+
+class CreateDeployment:
```

Comment:
Why is this a class? I feel like this could be a function of the `DeviceDeployment` class. It is used the same way you would use a function, and has a single public method. I feel like this is supposed to be a function. Of course, same goes for the `UpdateDeployment` and `DeleteDeployment` classes. that way, you also get rid of the global `NAME_SPACE`, as it actually belongs to your class.

Reply:
CreateDeployment, UpdatDeployment, DeleteDeployment will have more functionality to be expanded latter.
For e.g DeleteDeployment class can have below method:
a) CompleteCleanUp with directory as well
b) Delete only Deployments not meta-data or pvc data

UpdateDeployment:
a) ToolVersion check upgrade to stage wise upgrade based on local onboarding version
b) RollBack to version of onboarding you need and UpdateDeployment.

Reply:
Ok this make sense. I would suggest keeping in mind that merging those classes *long term* could still be a good idea, as they share lots of common functionality. (Maybe a inherit from a common parent class?)